.github/workflows/build.yml

```diff
@@ -173,3 +173,37 @@ jobs:
       uses: coverallsapp/github-action@v2
       with:
         parallel-finished: true
+
+  test-alpine:
+    runs-on: ubuntu-latest
+    needs:
+      - php-cs
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [ amd64, arm64 ]
+
+    steps:
+      - uses: actions/checkout@v4
+        name: Checkout repository
+      - name: Set up QEMU
+        if: matrix.arch == 'arm64'
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        if: matrix.arch == 'arm64'
+        uses: docker/setup-buildx-action@v3
+      # - name: Build and cache
+      #   uses: docker/build-push-action@v5
+      #   with:
+      #     context: .
+      #     push: false
+      #     tags: "alpine-base:${{ matrix.arch }}"
+      #     cache-from: type=gha
+      #     cache-to: type=gha,mode=max
+      #     platforms: linux/${{ matrix.arch }}
+      - name: run musl # If shared-mime-info not installed - Expected binary contents to have content type 'image/jpeg' but detected contents was 'application/octet-stream'
+        if: matrix.arch == 'amd64'
+        run: docker run --platform=linux/${{ matrix.arch }} --rm -v $PWD:/home you54f/pact-php:alpine-base-${{ matrix.arch }} /bin/sh -c 'apk add --no-cache shared-mime-info && cd /home && composer install && composer gen-lib && composer test'
```

Comment:
I can push these images to `pactfoundation` scope as an interim measure, otherwise I am happy to setup credentials in the repo to allow for publishing to DockerHub.

We could also publish to GitHub container registry, and then its nicely linked to the repo, and easy to setup creds. 

I am sure people would prefer having seen the images being published from a CI machine.

I noted there was an issue in loading the image from the cache in subsequent runs, which is why I ended up skipping the caching step

Reply:
Infact I think we can get away with a vanilla image and install some requirements directly from apk rather than building grpc from source

```
docker run --platform=linux/${{ matrix.arch }} --rm -v $PWD:/home alpine:3.19 /bin/sh -c 'apk add --no-cache shared-mime-info php82-dev php82-ffi php82-pecl-grpc php82-sockets php82-tokenizer php82-dom php82-xml php82-xmlwriter composer protoc protobuf-dev && cd /home && composer install && composer gen-lib && composer test'
```

Reply:
https://github.com/YOU54F/pact-php/blob/6d0e88b682377020e7b7b015fbe922ed744e4e58/.github/workflows/build.yml#L192

fixed up arm64 as it wasn't pulling the latest version of the csv plugin, so installing it manually in this run

https://github.com/pact-foundation/pact-plugins/issues/61

Need to update the manifest, and there wasn't a correct manifest file (which I've now added)

as a side note: does pact-php always try and grab the latest version of the plugin on the index, or can a user fix it to a particular version?

Reply:
passing run:- https://github.com/YOU54F/pact-php/actions/runs/9128336180

will get the plugin index updated today :)

Reply:
> pact-php always try and grab the latest version of the plugin on the index, or can a user fix it to a particular version?

pact-php doesn't have any control over which version of plugin to download

Reply:
hmmm, so the `pactffi_using_plugin` function takes two arguments, one is the version.

https://docs.rs/pact_ffi/latest/pact_ffi/plugins/fn.pactffi_using_plugin.html

What are we passing in there? It's nice to have it default to the latest (I think that it what is is doing if you pass in `NULL` to the 2nd arg)

however a user may want to fix, to a version they have installed on their local machine (to avoid trying to pull the latest) I'll check what happens if we have an earlier version of the plugin already installed, will it use that or just grab the latest.

From the provider verifier side, it looks at the pact file to determine which plugin to install. From what I've noted, every run will give me the latest plugins on the consumer side, and then the same on the provider side (as its reading from the regenned pacts)

Reply:
We can address the plugin version stuff in a different PR at another time rather than complicating this more

Reply:
> What are we passing in there? It's nice to have it default to the latest (I think that it what is is doing if you pass in NULL to the 2nd arg)

pact-php is using `NULL`

> I'll check what happens if we have an earlier version of the plugin already installed, will it use that or just grab the latest.

I remember I saw the behavior that it will use the version that's already installed, doesn't grab the latest.

Reply:
Thanks buddy.

Index now updated with the latest csv and avro plugins

https://github.com/pact-foundation/pact-plugins/pull/62